File: src/main/java/com/sandjelkovic/dispatchd/api/dto/EpisodeDTO.java
package com.sandjelkovic.dispatchd.api.dto;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.RequiredArgsConstructor;

import java.time.ZonedDateTime;

@Data
@RequiredArgsConstructor
@AllArgsConstructor
public class EpisodeDTO {

	private Long id;

	private ZonedDateTime airdate;

	@JsonIgnore
	private ZonedDateTime lastUpdated;

	private String customNumbering;

	private String description;

	private Integer number;

	private String seasonNumber;

	private String title;

	private String imdbId;

	private String tmdbId;

	private String traktId;

	private String tvdbId;

	private Long seasonId;

	private Long tvShowId;

	public EpisodeDTO id(final Long id) {
		this.id = id;
		return this;
	}

	public EpisodeDTO airdate(final ZonedDateTime airdate) {
		this.airdate = airdate;
		return this;
	}

	public EpisodeDTO lastUpdated(final ZonedDateTime lastUpdated) {
		this.lastUpdated = lastUpdated;
		return this;
	}

	public EpisodeDTO customNumbering(final String customNumbering) {
		this.customNumbering = customNumbering;
		return this;
	}

	public EpisodeDTO description(final String description) {
		this.description = description;
		return this;
	}

	public EpisodeDTO number(final Integer number) {
		this.number = number;
		return this;
	}

	public EpisodeDTO seasonNumber(final String seasonNumber) {
		this.seasonNumber = seasonNumber;
		return this;
	}

	public EpisodeDTO title(final String title) {
		this.title = title;
		return this;
	}

	public EpisodeDTO imdbId(final String imdbId) {
		this.imdbId = imdbId;
		return this;
	}

	public EpisodeDTO tmdbId(final String tmdbId) {
		this.tmdbId = tmdbId;
		return this;
	}

	public EpisodeDTO traktId(final String traktId) {
		this.traktId = traktId;
		return this;
	}

	public EpisodeDTO tvdbId(final String tvdbId) {
		this.tvdbId = tvdbId;
		return this;
	}

	public EpisodeDTO seasonId(final Long seasonId) {
		this.seasonId = seasonId;
		return this;
	}

	public EpisodeDTO tvShowId(final Long tvShowId) {
		this.tvShowId = tvShowId;
		return this;
	}
}

File: content-service/src/main/kotlin/com/sandjelkovic/dispatchd/contentservice/web/dto/EpisodeDto.kt
package com.sandjelkovic.dispatchd.contentservice.web.dto

import com.fasterxml.jackson.annotation.JsonIgnore
import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.sandjelkovic.dispatchd.contentservice.data.entity.Episode
import java.time.ZonedDateTime

/**
 * @author sandjelkovic
 * @date 2018-12-09
 */
@JsonIgnoreProperties(ignoreUnknown = true)
class EpisodeDto {

    private var id: Long? = null

    private var airdate: ZonedDateTime? = null

    @JsonIgnore
    private var lastUpdated: ZonedDateTime? = null

    private var customNumbering: String? = null

    private var description: String? = null

    private var number: Int? = null

    private var seasonNumber: String? = null

    private var title: String? = null

    private var imdbId: String? = null

    private var tmdbId: String? = null

    private var traktId: String? = null

    private var tvdbId: String? = null

    private var seasonId: Long? = null

    private var tvShowId: Long? = null

    fun id(id: Long?): EpisodeDto {
        this.id = id
        return this
    }

    fun airdate(airdate: ZonedDateTime): EpisodeDto {
        this.airdate = airdate
        return this
    }

    fun lastUpdated(lastUpdated: ZonedDateTime): EpisodeDto {
        this.lastUpdated = lastUpdated
        return this
    }

    fun customNumbering(customNumbering: String): EpisodeDto {
        this.customNumbering = customNumbering
        return this
    }

    fun description(description: String): EpisodeDto {
        this.description = description
        return this
    }

    fun number(number: Int?): EpisodeDto {
        this.number = number
        return this
    }

    fun seasonNumber(seasonNumber: String): EpisodeDto {
        this.seasonNumber = seasonNumber
        return this
    }

    fun title(title: String): EpisodeDto {
        this.title = title
        return this
    }

    fun imdbId(imdbId: String): EpisodeDto {
        this.imdbId = imdbId
        return this
    }

    fun tmdbId(tmdbId: String): EpisodeDto {
        this.tmdbId = tmdbId
        return this
    }

    fun traktId(traktId: String): EpisodeDto {
        this.traktId = traktId
        return this
    }

    fun tvdbId(tvdbId: String): EpisodeDto {
        this.tvdbId = tvdbId
        return this
    }

    fun seasonId(seasonId: Long?): EpisodeDto {
        this.seasonId = seasonId
        return this
    }

    fun tvShowId(tvShowId: Long?): EpisodeDto {
        this.tvShowId = tvShowId
        return this
    }
}

fun Episode.toDto() = Episode(
    id = id,
    version = version,
    airDate = airDate,
    lastUpdated = lastUpdated,
    customNumbering = customNumbering,
    description = description,
    title = title,
    number = number,
    seasonNumber = seasonNumber,
    imdbId = imdbId,
    tmdbId = tmdbId,
    traktId = traktId,
    tvdbId = tvdbId,
    imagesGroup = imagesGroup
)
